tests/test_learndot.py

```diff
@@ -342,3 +344,38 @@ def test_rate_limit_is_retried(self, status_code, retry, mock_logger):
             mock_logger.warning.assert_called_with(retry)
         else:
             mock_logger.warning.assert_not_called()
+
+
+class ModelMock:
+    def exists(self):
+        print('is this ok?')
+        return True
+
+
+class TestLearndotCommands(TestCase):
+    def _mock_edx_modules(self):
+        sys.modules['lms'] = MagicMock()
+        sys.modules['lms.djangoapps'] = MagicMock()
+        sys.modules['lms.djangoapps.courseware'] = MagicMock()
+        sys.modules['lms.djangoapps.courseware.courses'] = MagicMock()
+        sys.modules['lms.djangoapps.grades'] = MagicMock()
+        sys.modules['lms.djangoapps.grades.config'] = MagicMock()
+        sys.modules['lms.djangoapps.grades.course_grade_factory'] = MagicMock()
+        sys.modules['common'] = MagicMock()
+        sys.modules['common.djangoapps'] = MagicMock()
+        sys.modules['common.djangoapps.student'] = MagicMock()
+        sys.modules['common.djangoapps.student.models'] = MagicMock()
+
+    def setUp(self):
+        self.course_key = "course-v1:Test+TestCourse+201801"
+        self.user = User.objects.create(username="test", email="test@gmail.com", password="test")
+        self.client = LearndotAPIClientMock()
+        self._mock_edx_modules()
+        super(TestLearndotCommands, self).setUp()
+
+    @patch('lms.djangoapps.courseware.courses.get_course')
+    @patch('common.djangoapps.student.models.CourseEnrollment.objects')
+    def test_update_learndot_enrolments_with_date_range(self, *args):
+        from edxlearndot.management.commands.update_learndot_enrolments import Command
+        CourseMapping.objects.create(learndot_component_id=1, edx_course_key=self.course_key)
+        Command().handle(course_id=[self.course_key], start='two years ago', end='now', users=[])
```

Comment:
There's no verification happening here.. can you add an enrollment for that mapped course, and check:
* that with `--start="two years ago" --end="one year ago"` the enrollment is *not* processed
* that with `--start="two years ago" --end="now"` that the enrollment *is* processed?

You can check the `EnrolmentStatusLog` objects to verify.

Reply:
totally agree, it's improved now, please check it